Core/MIPS/ARM64/Arm64Asm.cpp

```diff
@@ -113,9 +115,109 @@ void Arm64Jit::GenerateFixedCode(const JitOptions &jo) {
 		loadStaticRegisters = nullptr;
 	}
 
-	enterCode = AlignCode16();
+	applyRoundingMode = AlignCode16();
 	if (!start)
-		start = enterCode;
+		start = applyRoundingMode;
+	{
+		LDR(INDEX_UNSIGNED, SCRATCH2, CTXREG, offsetof(MIPSState, fcr31));
+		if (!g_Config.bForceFlushToZero) {
+			TSTI2R(SCRATCH2, 1 << 24);
+			ANDI2R(SCRATCH2, SCRATCH2, 3);
+			FixupBranch skip1 = B(CC_EQ);
+			ADDI2R(SCRATCH2, SCRATCH2, 4);
+			SetJumpTarget(skip1);
+			// We can only skip if the rounding mode is zero and flush is set.
+			CMPI2R(SCRATCH2, 4);
+		} else {
+			ANDSI2R(SCRATCH2, SCRATCH2, 3);
+		}
+		// At this point, if it was zero, we can skip the rest.
+		FixupBranch skip = B(CC_EQ);
+		PUSH(SCRATCH1_64);
+
+		// MIPS Rounding Mode:       ARM Rounding Mode
+		//   0: Round nearest        0
+		//   1: Round to zero        3
+		//   2: Round up (ceil)      1
+		//   3: Round down (floor)   2
+		if (!g_Config.bForceFlushToZero) {
+			ANDI2R(SCRATCH1, SCRATCH2, 3);
+			CMPI2R(SCRATCH1, 1);
+		} else {
+			CMPI2R(SCRATCH2, 1);
+		}
+
+		FixupBranch skipadd = B(CC_NEQ);
+		ADDI2R(SCRATCH2, SCRATCH2, 2);
+		SetJumpTarget(skipadd);
+		FixupBranch skipsub = B(CC_LE);
+		SUBI2R(SCRATCH2, SCRATCH2, 1);
+		SetJumpTarget(skipsub);
+
+		MRS(SCRATCH1_64, FIELD_FPCR);
+		// Assume we're always in round-to-nearest mode beforehand.
+		if (!g_Config.bForceFlushToZero) {
+			// But we need to clear flush to zero in this case anyway.
```

Comment:
Right.  Well, I think `bForceFlushToZero` is not doing anything now here.  It would have to `ORR` in `4 << 22`.

When restoring, I guess we should just not `ORR` it in?  It seems like this would currently re-enable flush to zero even if C++ turned it off, unless C++ is saving the state somewhere?

Maybe we just remove `g_Config.bForceFlushToZero` if it breaks things and doesn't have a big enough perf impact to be useful?

-[Unknown]